src/auth/jwtclient.ts

```diff
@@ -93,11 +98,26 @@ export class JWT extends OAuth2Client {
    */
   protected async getRequestMetadataAsync(url?: string|null):
       Promise<RequestMetadataResponse> {
-    if (this.createScopedRequired() && url) {
-      // no scopes have been set, but a uri has been provided.  Use JWTAccess
-      // credentials.
-      const alt = new JWTAccess(this.email, this.key);
-      return alt.getRequestMetadata(url);
+
+    if (!this.apiKey && this.createScopedRequired() && url) {
+      // If there's no API key, no scopes, and a URL is included,
+      // assume we're using a JWT based workflow.
+      if (this.additionalClaims && (this.additionalClaims as any).target_audience) {
+        // The `target_audience` claim is a special claim supported by the
+        // /oauth2/v4/token endpoint and specifically IAP. If that claim is
+        // included, the token endpoint will return an id_token instead of
+        // an access_token.
+        const {tokens} = await this.refreshToken();
+        return {headers: {Authorization: `Bearer ${tokens.access_token}`}};
+      } else {
+        // TODO: There is some duplication between the code in google/node-gtoken
+        // and the code in JWTAccess.  This particular code path needs to be
+        // further rationalized.
+        if (!this.access) {
```

Comment:
Why does this value now have to be cached? (Is that what you meant by 'needs to be further rationalized'?)

Reply:
Hah, that's a great question.  So in #254 I added caching inside of the JWTAccess object for gtokens.  However - we create a new instance of that object every time `getRequestMetadata` is called, so the cache does nothing 😂 . This actually enables the cache.